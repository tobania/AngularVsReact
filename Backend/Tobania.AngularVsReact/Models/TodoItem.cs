
using System;
namespace Tobania.AngularVsReact.Models
{
    public class TodoItem
    {
        public Guid Id { get; set; }
        public string Text { get; set; }
        public bool Checked { get; set; }

        public Todo Parent { get; set; }

        public TodoItem() { }

        public TodoItem(Todo parent, string text)
        {
            Id = Guid.NewGuid();
            Text = text;
            Parent = parent;
        }

        public void Update(string text, bool isChecked = false)
        {
            Text = text;
            Checked = isChecked;
        }
    }
}