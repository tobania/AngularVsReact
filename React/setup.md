# React Setup

## Prerequirements
- nodejs
- Backend runs in the background and is accessable

# Setup
Navigate to the frontend root (src) and execute the following commands:
npm install
npm start
