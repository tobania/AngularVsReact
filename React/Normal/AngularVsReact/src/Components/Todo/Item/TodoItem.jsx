import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import toggleTodoItemService from '../../../HttpServices/TodoItems/ToggleTodoItemService';

class TodoItem extends PureComponent {
  constructor() {
    super();
    this.toggle = this.toggle.bind(this);
  }

  toggle() {
    const { todoId, identifier, toggleTodoItem } = this.props;
    toggleTodoItemService(todoId, identifier, toggleTodoItem);
  }

  render() {
    const { identifier, text, checked } = this.props;
    return (
      <li className="list-group-item">
        <div className="form-check">
          <input type="checkbox" className="form-check-input" id={identifier} checked={checked} onChange={this.toggle} />
          <label className="form-check-label" htmlFor={identifier}>{text}</label>
        </div>
      </li>
    );
  }
}

TodoItem.propTypes = {
  todoId: PropTypes.string.isRequired,
  identifier: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  checked: PropTypes.bool,
  toggleTodoItem: PropTypes.func.isRequired,
};

TodoItem.defaultProps = {
  checked: false,
};

export default TodoItem;
