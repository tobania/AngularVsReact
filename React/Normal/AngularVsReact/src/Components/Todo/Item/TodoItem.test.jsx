import React from 'react';
import renderer from 'react-test-renderer';
import TodoItem from './TodoItem';

describe('TodoItem', () => {
  it('gets loaded correctly', () => {
    const component = renderer.create(
      <TodoItem todoId="todoId" identifier="identifier" text="text" toggleTodoItem={jest.fn} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('toggle can be called', () => {
    const component = renderer.create(
      <TodoItem todoId="todoId" identifier="identifier" text="text" toggleTodoItem={jest.fn} />,
    ).getInstance();

    component.toggle();
  });
});
