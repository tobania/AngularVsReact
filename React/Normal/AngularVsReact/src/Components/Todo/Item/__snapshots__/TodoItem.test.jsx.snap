// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem gets loaded correctly 1`] = `
<li
  className="list-group-item"
>
  <div
    className="form-check"
  >
    <input
      checked={false}
      className="form-check-input"
      id="identifier"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="form-check-label"
      htmlFor="identifier"
    >
      text
    </label>
  </div>
</li>
`;
